From 8102ff4041da74ff25645b249b291797548126bf Mon Sep 17 00:00:00 2001
From: Yunjie Hu <yh1112@nyu.edu>
Date: Mon, 7 Dec 2015 13:19:08 -0500
Subject: [PATCH 2/2] Add password protection

---
 init.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/init.c b/init.c
index 0392c17..45ffea7 100644
--- a/init.c
+++ b/init.c
@@ -25,6 +25,9 @@ main(void)
   }
   dup(0);  // stdout
   dup(0);  // stderr
+  if (open("pass.txt", O_RDONLY) != -1){
+      pFlag = 1;
+  }
   if (pFlag ==0){
     printf(1, "No password set. Please choose one.\n");
   }
@@ -34,8 +37,8 @@ main(void)
        printf(1,"Re-enter to confirm: ");
        read(0,b, sizeof(b));
         if (strcmp(a , b) != 0){
-            pFlag = 0;
             printf(1, "Passwords do not match. Try again.\n");
+            pFlag = 0;
         }
         else{
             uint randa = random();
-- 
2.5.3

